Signal/src/util/DateUtil.m

```diff
@@ -65,21 +68,34 @@ + (BOOL)dateIsToday:(NSDate *)date {
     return [self date:[NSDate date] isEqualToDateIgnoringTime:date];
 }
 
-+ (NSString *)formatPastTimestampRelativeToNow:(uint64_t)pastTimestamp
++ (BOOL)dateIsYesterday:(NSDate *)date
+{
+    NSDate *yesterday = [NSDate ows_dateWithMillisecondsSince1970:[NSDate ows_millisecondTimeStamp] - kDayInMs];
+    return [self date:yesterday isEqualToDateIgnoringTime:date];
+}
+
++ (NSString *)formatPastTimestampRelativeToNow:(uint64_t)pastTimestamp isRTL:(BOOL)isRTL
 {
     OWSCAssert(pastTimestamp > 0);
 
     uint64_t nowTimestamp = [NSDate ows_millisecondTimeStamp];
     BOOL isFutureTimestamp = pastTimestamp >= nowTimestamp;
 
     NSDate *pastDate = [NSDate ows_dateWithMillisecondsSince1970:pastTimestamp];
+    NSString *dateString;
     if (isFutureTimestamp || [self dateIsToday:pastDate]) {
-        return [[self timeFormatter] stringFromDate:pastDate];
+        dateString = NSLocalizedString(@"DATE_TODAY", @"The current day.");
+    } else if ([self dateIsYesterday:pastDate]) {
+        dateString = NSLocalizedString(@"DATE_YESTERDAY", @"The day before today.");
     } else if (![self dateIsOlderThanOneWeek:pastDate]) {
-        return [[self weekdayFormatter] stringFromDate:pastDate];
+        dateString = [[self weekdayFormatter] stringFromDate:pastDate];
```

Comment:
Do you think the day of week formatting is helpful? 

I'd prefer to get rid of this block altogether and only ever see "Today", "Yesterday", or "2/12/18". And never see (e.g.) "Tuesday".

Reply:
I like it actually, but don't care much.  Removed!